docs/source/inprocess/usage_guide.rst

```diff
@@ -377,6 +377,55 @@ There are two methods to record progress:
 Timeout event is triggered if either of the active progress monitoring methods
 didn't record a heartbeat in the specified time interval.
 
+Disabling hang protection
+^^^^^^^^^^^^^^^^^^^^^^^^^
+In some cases, certain operations within the wrapped function may legitimately
+take longer than the configured ``soft_timeout`` or ``hard_timeout`` intervals.
+For such operations, the :py:class:`Wrapper` provides a
+:py:meth:`inprocess.CallWrapper.disable_hang_protection` context manager that
+temporarily disables timeout-based hang detection.
+
+.. warning::
+    The :py:meth:`inprocess.CallWrapper.disable_hang_protection` context manager
+    disables critical safety mechanisms designed to protect against hangs and
+    deadlocks. Use with extreme caution and only for operations you are confident
+    will complete.
+
+The context manager is typically used for operations such as:
+
+- Large data loading operations that may take unpredictable amounts of time
+- Complex initialization routines
+- Other legitimate long-running operations that should not be interrupted
+
+When the context manager exits (either normally or due to an exception), hang
+protection is automatically re-enabled.
+
+.. code-block:: python
+
+    def my_training_function(call_wrapper: CallWrapper):
+        # Normal operations are subject to hang protection
+        train_step()
+
+        # Disable hang protection for long-running data loading
```

Comment:
I think this is a good first step, but I'm worried about cases seen recently where the storage cluster was being thrashed with `getattr()` calls and hung for a very long time.

Perhaps in the future we could add a way to temporarily extend the timeout for a section instead of disabling it completely.